cns/kubecontroller/nodenetworkconfig/conversion.go

```diff
@@ -86,15 +90,44 @@ func CreateNCRequestFromStaticNC(nc v1alpha.NetworkContainer) (*cns.CreateNetwor
 	}
 
 	secondaryIPConfigs := map[string]cns.SecondaryIPConfig{}
+	if nc.Type == v1alpha.VNETBlock {
+		// iterate through all IP addresses in the subnet described by primaryPrefix and
+		// add them to the request as secondary IPConfigs.
+		// Skip first IP in the primary ip CIDR block during secondary ip assignments
+		zeroAddr := primaryPrefix.Masked().Addr().Next()
+		for addr := zeroAddr; primaryPrefix.Contains(addr); addr = addr.Next() {
+			secondaryIPConfigs[addr.String()] = cns.SecondaryIPConfig{
+				IPAddress: addr.String(),
+				NCVersion: int(nc.Version),
+			}
+		}
```

Comment:
This looks suspiciously like the entire contents of the `else` case. What if you pulled that code out in front of the `if` block, since it needs to happen in both branches? Then you can drop the else entirely, and there's less code in the `if`. Significantly simpler control flow, imo.

Reply:
Sounds fine to me.
Only thing is, since in VNETBlock mode we exclude first IP from primaryIP CIDR and dont mask primaryIP,  I will have to use two if statements. One for  startAddr initialization and one for IP assignments. from CIDR blocks